packages/monitor-v2/src/bot-oo/SettleOOv2Requests.ts

```diff
@@ -0,0 +1,104 @@
+import { paginatedEventQuery } from "@uma/common";
+import {
+  RequestPriceEvent,
+  SettleEvent,
+} from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/OptimisticOracleV2";
+import { ethers } from "ethers";
+import { computeEventSearch } from "../bot-utils/events";
+import { logSettleRequest } from "./BotLogger";
+import { getContractInstanceWithProvider, Logger, MonitoringParams, OptimisticOracleV2Ethers } from "./common";
+import { requestKey } from "./requestKey";
+
+export async function settleOOv2Requests(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const oo = await getContractInstanceWithProvider<OptimisticOracleV2Ethers>("OptimisticOracleV2", params.provider);
+
+  const searchConfig = await computeEventSearch(
+    params.provider,
+    params.blockFinder,
+    params.timeLookback,
+    params.maxBlockLookBack
+  );
+
+  const requests = await paginatedEventQuery<RequestPriceEvent>(oo, oo.filters.RequestPrice(), searchConfig);
+
+  const settlements = await paginatedEventQuery<SettleEvent>(oo, oo.filters.Settle(), searchConfig);
+
+  const settledKeys = new Set(settlements.map((e) => requestKey(e.args)));
+
+  const requestsToSettle = requests.filter((e) => !settledKeys.has(requestKey(e.args))) as RequestPriceEvent[];
```

Comment:
Isn't it easier to start with proposed events? Request might be too far back in the past (longer than lookback), so we could miss quite a lot of settleable proposals. Same comment applies to OOv1